torch_xla/experimental/pallas_kernels/quantized_matmul_kernel.py

```diff
@@ -14,69 +14,116 @@ def _quantize_array(
   n_bits = 8
   int_max = 2**(n_bits - 1) - 1
   scale = (x_abs_max_val / int_max).T  # [bs_block_size, 1]
-  # Need to explicitly cast to f32 because Mosaic can't directly jnp.round a
-  # bf16 array.
-  # It seems x/0 in Pallas generates inf/-inf instead of an exception.
-  x_int = jnp.round((x / scale).astype(jnp.float32)).astype(jnp.int8)
-  return x_int, scale.astype(x.dtype)
+  x_int = jnp.round(x / scale).astype(jnp.int8)
+  return x_int, scale.astype(jnp.float32)
```

Comment:
What's the reason to change this `scale.astype(x.dtype)` to `scale.astype(jnp.float32)`?

Reply:
It's because `scale` will eventually need to be converted into `float32` during dequantization. so there is no reason to convert `scale` into `x.dtype` (which is usually `bf16`) and convert it into `float32` again. Better to just straight up convert it into `float32`.